### PR TITLE
Copy backup button

### PR DESCRIPTION
**Pull request by Grayyyyy**
https://github.com/lbryio/lbry-app/issues/1635 completely implemented with this. Needs a bit of testing; worked on my machine.

### CODE REVIEW

**Comment by seanyesmunt** [comment]
Instead of creating a component specific backup copy, we should create a more generic component. This would be used by the address component, and on the backup page. Maybe something like this:

```
<CopyableInput
    inputText={lbryumWalletDir}
    snackbarText={__('Folder path copied')}
    showCopyButton
/>
```

Then we aren't duplicating very similar code accross the app.

**Comment by seanyesmunt** [comment]
I closed the other PR since the code exists in this PR. But those comments apply to this one as well.

**Comment by Grayyyyy** [comment]
Alright, I'll try.

**Comment by Grayyyyy** [comment]
Alright, I've created a new component called CopyableText. It's basically a universal component for making text boxes that are copyable, not just for Backup and Address. Clicking the copy button brings up a popup saying "Text copied" and copies the text box into the clipboard. Was this what you requested?

**Comment by seanyesmunt** [comment]
Yep that is what I was thinking. This would be used in the backup copy, shapeshift, and wallet address. Can you push up these changes so I can review them?

**Comment by kauffj** [comment]
@Grayyyyy time for you to learn a new skill!

Can you `rebase` and `squash` these changes so that they come in as a single commit on top of the current `master` branch?

https://github.com/lbryio/lbry/wiki/Branching-and-Merging#rebase-onto-master-daily

(If you find this tricky, we can accept this as-is and do it ourselves, but I figured you'd want to learn)

File: src/renderer/component/copyableText/view.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+          }}
+        />
+        <Button
+          noPadding
+          button="secondary"
+          icon={icons.CLIPBOARD}
+          onClick={() => {
+            clipboard.writeText(copyable);
+            doNotify({
+              message: __('Text copied'),
```
Comment (seanyesmunt) [inline src/renderer/component/copyableText/view.jsx]: We can default to `Text copied` but we should be able to pass in any message to use. Ex: on the wallet overview/shapeshift, we should say `Address copied`.

**Comment by Grayyyyy** [comment]
Just rebased and squashed... I think. GitHub is saying "Changes requested", how do I approve the change you made to the code?

**Comment by seanyesmunt** [comment]
I have to approve it 🙂 

Can you push up your changes after rebasing? You will have to do a force push

**Comment by Grayyyyy** [comment]
Rebased and force pushed my changes.

**Comment by Grayyyyy** [comment]
Do I need to add your code change to the code manually?

**Comment by seanyesmunt** [comment]
No I will push a commit to your PR. It looks like this component still doesn't accept a prop that changes the notification message. Could you add that? See my comment above.

**Comment by Grayyyyy** [comment]
Hey, I don't know how to do this, since while I do know a bit of JavaScript, I don't know ReactJS. I couldn't figure out how to add the "custom message/prop" thing. Are there any resources/tips that would be helpful?

File: src/renderer/component/shapeShift/internal/active-shift.jsx [hunk 2]
```diff
@@ -94,7 +94,7 @@ class ActiveShapeShift extends React.PureComponent<Props> {
             {shiftDepositAddress && (
               <FormRow verticallyCentered padded>
                 <QRCode value={shiftDepositAddress} paddingRight />
-                <Address address={shiftDepositAddress} showCopyButton padded />
+                <CopyableText copyable={shiftDepositAddress} showCopyButton padded />
```
Comment (seanyesmunt) [inline src/renderer/component/shapeShift/internal/active-shift.jsx]: Just realized you are calling this prop `copyable`. A better name would be `value`.

**Comment by seanyesmunt** [comment]
Hey @Grayyyyy 

The official react docs are pretty good. https://reactjs.org/docs/components-and-props.html

This new prop `notificationMessage` will be passed in from the component that use `CopyableText`

It could work like this:

```
// address/view.jsx
  ..
  <CopyableText notificationMessage={__('Address copied')} value={walletAddress} />
```

Then in the component, you might have something like this:
```
// copyable-text.jsx
export function (props) => {
  const { doNotify, notificationMessage } = props;
  return (
    ...
    <Button onClick={() => doNotify({
         message: notificationMessage || __('Text copied'),
         displayType: ['snackbar'],
    )} />
  )
}
```

You are just

**Comment by seanyesmunt** [comment]
Just realized I commented midway through typing a sentence. Just completed my thought.

**Comment by seanyesmunt** [comment]
@Grayyyyy are you planning on finishing this up? Let me know if you have additional questions, maybe we could chat over discord.

**Comment by Grayyyyy** [comment]
I am planning on finishing this up, and if I have any questions, I'll ping you on Discord 👍

**Comment by tzarebczan** [comment]
@Grayyyyy any updates?

**Comment by seanyesmunt** [comment]
I'm going to close this. @Grayyyyy If you plan to finish this up you can re-open and continue to push commits to your branch.

**Pull request closed without merging**